src/pages/Storefront/Cart.js

```diff
@@ -14,23 +17,102 @@ import {
   Button,
   Input,
   Link,
+  Tag,
+  TagLabel,
+  TagLeftIcon,
+  TagCloseButton,
   chakra,
 } from '@chakra-ui/react';
-import PreserveQueryParamsLink from 'components/storefront/PreserveQueryParamsLink/PreserveQueryParamsLink';
-import { PageContainer } from 'components/storefront/PageContainer/PageContainer';
-import CartSkeleton from 'components/storefront/Cart/Layout/CartSkeleton';
+import { ImPriceTags } from 'react-icons/im';
+
+const calculateDiscount = (discountApplications, lineItems) => {
+  let discountVal = 0;
+  let shippingCost = 'TBD';
+  let discountType = '';
+  if (discountApplications.length > 0) {
+    const appliedDiscount = discountApplications[0];
+    discountType = appliedDiscount.targetType;
+    if (discountType === 'SHIPPING_LINE') {
+      shippingCost = 'FREE';
+    } else if (discountType === 'LINE_ITEM') {
+      discountVal = lineItems
+        .reduce((acc, lineItem) => {
+          return lineItem.discountAllocations
+            ? acc +
+                lineItem.discountAllocations.reduce((acc2, discAlloc) => {
+                  return acc2 + parseFloat(discAlloc.allocatedAmount.amount);
+                }, 0)
+            : acc;
+        }, 0)
+        .toFixed(2);
+    }
+  }
+  return { discountVal, shippingCost, discountType };
+};
+
+const DiscountMsg = ({ discountApplications, discountSuccess, removeCoupon }) => {
+  const msg =
+    discountSuccess === 'SUCCESS' ? (
+      <chakra.h4 whiteSpace="nowrap" textStyle="lightCaption">
+        Success! Your code <b>{discountApplications[0].code}</b> has been applied.
+      </chakra.h4>
+    ) : discountSuccess === 'INVALID' ? (
+      <chakra.h4 color="brand.red" textStyle="lightCaption">
+        Sorry, that coupon code is not applicable.
+      </chakra.h4>
+    ) : null;
+
+  return (
+    <Box pt="4">
+      {msg}
+      {discountApplications && discountApplications[0] && (
+        <Tag size="lg" mt="2">
+          <TagLeftIcon boxSize="12px" as={ImPriceTags} />
+          <TagLabel>{discountApplications[0].code}</TagLabel>
+          <TagCloseButton onClick={removeCoupon} />
+        </Tag>
```

Comment:
oh yeah, one bug you had was that it wouldn't show the coupon code if u refreshed the page/navigated pages, but coding it this way shows the coupon code always 👍